### PR TITLE
Add query parameters supported by CC 3.165.0 list operations

### PR DESCRIPTION
**Pull request by freudl**
Listing resources supports query parameters named in docs.

* `AppUsageClient` will be able to [list events](http://v3-apidocs.cloudfoundry.org/version/3.165.0/index.html#list-apps) `after_guid`
* Further params in `package`, `process`, `stack` and `task`

### CODE REVIEW

File: client/process.go [hunk 1]
```diff
@@ -14,12 +14,14 @@ type ProcessClient commonClient
 type ProcessListOptions struct {
 	*ListOptions
 
-	GUIDs             Filter `qs:"guids"`
-	Types             Filter `qs:"types"`
-	Names             Filter `qs:"names"`
```
Comment (freudl) [inline client/process.go:19]: Drop `names` as this is [no query parameter](http://v3-apidocs.cloudfoundry.org/version/3.165.0/index.html#list-processes)

**Comment by freudl** [comment]
Query parameter [`fields`](http://v3-apidocs.cloudfoundry.org/version/3.165.0/index.html#resources-with-fields) remains unsupported.

**Comment by sneal** [comment]
Thanks for the PR @freudl! Is there a reason you duplicated [some of the fields from ListOptions](https://github.com/cloudfoundry/go-cfclient/blob/a5dc9be0ac6ba88cb10d672cc2bb4bc7c0356df2/client/list_opt.go#L35-L37) into each specific option type? Typical usage for example:

```go
opt := client.NewAppListOptions()
opts.CreateAts.EqualTo(mydate)
```

**Comment by freudl** [comment]
Thanks for pointing out, @sneal I removed the redundant fields. A few fields remain, still.

**Comment by sneal** [comment]
LGTM, thanks!

**Pull request merged**